Source/WebCore/rendering/style/RenderStyle.cpp

```diff
@@ -2279,20 +2283,26 @@ Color RenderStyle::colorResolvingCurrentColor(CSSPropertyID colorProperty, bool
         }();
 
         if (!visitedLink && (borderStyle == BorderStyle::Inset || borderStyle == BorderStyle::Outset || borderStyle == BorderStyle::Ridge || borderStyle == BorderStyle::Groove))
-            return SRGBA<uint8_t> { 238, 238, 238 };
+            return { SRGBA<uint8_t> { 238, 238, 238 } };
 
-        return visitedLink ? visitedLinkColor() : color();
+        return colorCurrentColor(visitedLink);
     }
 
-    return result;
+    return colorResolvingCurrentColor(result);
+    
 }
 
-Color RenderStyle::colorResolvingCurrentColor(const Color& color) const
+Color RenderStyle::colorResolvingCurrentColor(const StyleColor& color) const
 {
-    if (isCurrentColor(color))
-        return this->color();
+    ColorContext colorContext;
+    colorContext.currentColor = this->color().resolveColor();
```

Comment:
This should be able to use absoluteColor() as 'currentcolor' for color property gets translated in style builder.